zuul-core/src/main/java/com/netflix/zuul/netty/server/Server.java

```diff
@@ -287,6 +301,17 @@ public void uncaughtException(final Thread t, final Throwable e) {
             Runtime.getRuntime().addShutdownHook(new Thread(() -> stop(), "Zuul-ServerGroup-JVM-shutdown-hook"));
         }
 
+        synchronized List<SocketAddress> getListeningAddresses() {
```

Comment:
s/List/Set ?

Reply:
I don't think this can be, since equality on SocketAddresses is not well defined.   (resolved InetSocketAddresses vs unresolved ones)